pkgs/applications/graphics/processing/default.nix

```diff
@@ -1,73 +1,100 @@
   pname = "processing";
-  version = "3.5.4";
+  version = "4.1.1";
 
   src = fetchFromGitHub {
     owner = "processing";
-    repo = "processing";
-    rev = "processing-0270-${version}";
-    sha256 = "0cvv8jda9y8qnfcsziasyv3w7h3w22q78ihr23cm4an63ghxci58";
+    repo = "processing4";
+    rev = "processing-${buildNumber}-${version}";
+    sha256 = "sha256-OjTqANxzcW/RrAdqmVYAegrlLPu6w2pjzSyZyvUYIt4=";
   };
 
-  patches = [
-    (fetchpatch {
-      name = "oraclejdk-8u281-compat.patch";
-      url = "https://github.com/processing/processing/commit/7e176876173c93e3a00a922e7ae37951366d1761.patch";
-      sha256 = "g+zwpoIVgw7Sp6QWW3vyPZ/fKHk+o/YCY6xnrX8IGKo=";
-    })
-  ];
+  nativeBuildInputs = [ ant unzip makeWrapper wrapGAppsHook ];
+  buildInputs = [ jdk javaPackages.jogl_2_3_2 ant rsync ffmpeg batik ];
 
-  nativeBuildInputs = [ ant rsync makeWrapper ];
-  buildInputs = [ jdk ];
+  dontWrapGApps = true;
 
   buildPhase = ''
-    # use compiled jogl to avoid patchelf'ing .so files inside jars
-    rm core/library/*.jar
-    cp ${javaPackages.jogl_2_3_2}/share/java/*.jar core/library/
-
-    # do not download a file during build
-    ${xmlstarlet}/bin/xmlstarlet ed --inplace -P -d '//get[@src="http://download.processing.org/reference.zip"]' build/build.xml
-    install -D -m0444 ${fetchurl {
-                          # Use archive.org link for reproducibility until the following issue is fixed:
-                          # https://github.com/processing/processing/issues/5711
-                          url = "https://web.archive.org/web/20200406132357/https://download.processing.org/reference.zip";
-                          sha256 = "093hc7kc9wfxqgf5dzfmfp68pbsy8x647cj0a25vgjm1swi61zbi";
-                        }
-                       } ./java/reference.zip
-
-    # suppress "Not fond of this Java VM" message box
-    substituteInPlace app/src/processing/app/platform/LinuxPlatform.java \
-      --replace 'Messages.showWarning' 'if (false) Messages.showWarning'
-
-    ( cd build
-      substituteInPlace build.xml --replace "jre-download," ""  # do not download jre1.8.0_144
-      mkdir -p linux/jre1.8.0_144                               # fake dir to avoid error
-      ant build )
+    echo "tarring jdk"
+    tar --checkpoint=10000 -czf build/linux/jdk-17.0.5-amd64.tgz ${jdk}
+    cp ${ant}/lib/ant/lib/{ant.jar,ant-launcher.jar} app/lib/
+    mkdir -p core/library
+    ln -s ${javaPackages.jogl_2_3_2}/share/java/* core/library/
+    ln -s ${vaqua} app/lib/VAqua9.jar
+    ln -s ${flatlaf} app/lib/flatlaf.jar
+    ln -s ${lsp4j} java/mode/org.eclipse.lsp4j.jar
+    ln -s ${lsp4j-jsonrpc} java/mode/org.eclipse.lsp4j.jsonrpc.jar
+    ln -s ${gson} java/mode/gson.jar
+    unzip -qo ${jna} -d app/lib/
+    mv app/lib/{jna-5.10.0/dist/jna.jar,}
+    mv app/lib/{jna-5.10.0/dist/jna-platform.jar,}
+    ln -sf ${batik}/* java/libraries/svg/library/
+    cp java/libraries/svg/library/lib/batik-all-1.15.jar java/libraries/svg/library/batik.jar
```

Comment:
```
tarring jdk
tar: Removing leading `/' from member names
tar: Write checkpoint 10000
tar: Write checkpoint 20000
tar: Write checkpoint 30000
tar: Write checkpoint 40000
tar: Write checkpoint 50000
cp: cannot stat 'java/libraries/svg/library/lib/batik-all-1.15.jar': No such file or directory
```

Reply:
Fixed, strange that it built fine for me. Out of curiosity, what command do you run to build from your local nixpkgs? I have been doing:

```
nix-env -f ~/git/nixpkgs -iA processing
```

but I have noticed it doesn't always do a new build, even if I have changed the `.nix` file. The following seems to always do a build:

```
nix-build ~/git/nixpkgs -A processing --check
```

Reply:
I always do ``nix-build -A $NAME``